Community Toolbox/scripts/tests_ArrayMaxTests/tests_ArrayMaxTests.gml

```diff
@@ -19,19 +24,60 @@ function ArrayMaxTests(_run, _method) : VerrificMethodTest(_run, _method) constr
         then_result().should_be(10);
     }
     
+    static should_handle_large_array = function() {
+        given_array(array_with_100k_items);
+        when_array_max_runs();
+        then_result().should_be(99_999);
+    }
+    
+    static should_handle_positive_offset_and_length = function() {
+        given_array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+        given_offset_and_length(3, 4); // should take [4, 5, 6, 7]
+        when_array_max_runs();
+        then_result().should_be(7);
+    }
+    
+    static should_handle_negative_offset = function() {
+        given_array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+        given_offset_and_length(-5, 3); // should take [6, 7, 8]
+        when_array_max_runs();
+        then_result().should_be(8);
+    }
+    
+    static should_handle_negative_length = function() {
+        given_array([1, 2, 3, 4, 5, 6, 7, 8, 9, 10]);
+        given_offset_and_length(5, -3); // should take [6, 5, 4]
+        when_array_max_runs();
+        then_result().should_be(6);
+    }
+    
+    static should_handle_large_subsection = function() {
+        given_array(array_with_100k_items);
+        given_offset_and_length(20_000, 70_000); // should take [20000, 20001, ..., 89998, 89999]
+        when_array_max_runs();
+        then_result().should_be(89_999);
+    }
+    
     // -----
     // Setup
     // -----
     
     array = [];
+    offset = 0;
+    length = undefined;
```

Comment:
Also, whether or not you consider the above to be important, I recommend adding at least one test that passes offset and not length.

Reply:
Updated.